analyzer/src/main/kotlin/managers/utils/MavenSupport.kt

```diff
@@ -306,19 +306,21 @@ class MavenSupport(private val workspaceReader: WorkspaceReader) {
          * Return true if an artifact that has not been requested from Maven Central is also available on Maven Central
          * but with a different hash, otherwise return false.
          */
-        private fun isArtifactModified(artifact: Artifact, remoteArtifact: RemoteArtifact): Boolean {
-            if (remoteArtifact.url.isBlank() || isMavenCentralUrl(remoteArtifact.url)) return false
-
-            val mavenCentralUrl = with(artifact) {
-                val group = groupId.replace('.', '/')
-                val name = remoteArtifact.url.substringAfterLast('/')
-                val hash = remoteArtifact.hash.algorithm.name.lowercase()
-                "https://repo.maven.apache.org/maven2/$group/$artifactId/$version/$name.$hash"
-            }
+        private fun isArtifactModified(artifact: Artifact, remoteArtifact: RemoteArtifact): Boolean =
+            with(remoteArtifact) {
+                if (url.isBlank() || isMavenCentralUrl(url)) return false
 
-            val checksum = OkHttpClientHelper.downloadText(mavenCentralUrl).getOrElse { return false }
-            return !remoteArtifact.hash.verify(parseChecksum(checksum, remoteArtifact.hash.algorithm.name))
-        }
+                val name = url.substringAfterLast('/')
+                val algorithm = hash.algorithm.name.lowercase()
+
```

Comment:
Having two `with` nested it gets a bit harder to see from which object the attributes come from.

Would it make sense to solve this by factoring out `fun Artifact.getMavenCentralUrl()?`

Reply:
> Would it make sense to solve this by factoring out `fun Artifact.getMavenCentralUrl()`?

That's not nicely possible, as the function would also need to have access to the `RemoteArtifact`, and I wouldn't want to pass that as an argument.

Given that the inner `with` is only two lines, and with the IDE you get proper indication about the scope of the properties, can we live with the nested `with` in this case?

Reply:
Yes, my comment didn't make sense as I tried to explain below.